workflows/fossa-scan.yml

```diff
@@ -0,0 +1,25 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2025 Canonical Ltd.
+name: Fossa Scan
+
+on:
+  workflow_call:
+    inputs:
+      branch-name:
+        description: Name of the branch to checkout
+        required: false
+        type: string
+        default: ${{ github.ref }}
+
+jobs:
+  fossa-scan:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.branch-name }}
+
+      - name: FOSSA scan
+        uses: fossa-contrib/fossa-action@v3
+        with:
+          fossa-api-key: 0c3bbcdf20e157bbd487dae173751b28
```

Comment:
Should this be a secret?

Reply:
No, this cannot be a secret because secrets are not shared as part of PRs. The `fossa-api-key` is a push-only token as suggested in the GHA: https://github.com/marketplace/actions/fossa-action#push-only-api-token

Reply:
Ok, thanks for the confirmation.